libtelemetry/src/main/java/com/mapbox/android/telemetry/TelemetryService.java

```diff
@@ -287,8 +291,21 @@ private boolean locationPermissionCheck() {
       int finePermission = PermissionChecker.checkSelfPermission(MapboxTelemetry.applicationContext,
         Manifest.permission.ACCESS_FINE_LOCATION);
 
+      if (checkFinePermission(finePermission)) {
+        return false;
+      }
+
       return coarsePermission == PackageManager.PERMISSION_GRANTED
```

Comment:
does it actually make sense to check for coarse permission with passive provider at all?